### ISSUE TITLE
The results of the “8 input views” lego did not match the paper

### ISSUE THREAD

**Issue by F011011110100** [Fictionarry/DNGaussian#40]
Hello!
I ran your work, but I ran it multiple times and the result was worse than the result of the source 3DGS. I used 200 sets of test and 8 sets of train to run DNGaussian and 3DGS and iterated 30k times, and their initial point clouds were obtained after sampling 100,000 points through lego mesh truth value. The following result paper(1) uses the same command as 3DGS, "`python train_blender.py -s data/lego_s_8 --m output/ --eval`", paper(2) and paper(3) are the results of two runs using the lego part of the script in the scripts you provided.  (with slightly modified iterations"`
python train_blender.py -s data/nerf_synthetic/sparse_views/lego_s_8 --model_path output/paper -r 2 --eval --lambda_dssim 0.2 \
            --densify_grad_threshold 0.0002 --prune_threshold 0.005 --percent_dense 0.01 \
            --densify_from_iter 500 \
            --position_lr_init 0.00016 --position_lr_final 0.0000016 --position_lr_start 5000 \
            --hard_depth_start 0 \
            --error_tolerance 0.01 \
            --scaling_lr 0.005 \
            --shape_pena 0.000 --opa_pena 0.000 --scale_pena 0.000 \
            --use_SH
`") . These three results are all inferior to those of 3DGS. Is there any possible reason for this?
- paper(1) 22.72(6k steps) 21.74(30k steps)
- paper(2) 24.06(6k steps) 23.96(30k steps)
- paper(3) 24.29(6k steps) 23.79(30k steps)
- 3dgs 24.37(6k steps) 24.25(30k steps)
Looking forward to your reply！

**Comment by Fictionarry** [comment]
Hi, the main problem may be here “initial point clouds were obtained after sampling 100,000 points through lego mesh truth value”. In other words, the initial point cloud has already contained ideal ground truth information that can not be fully gained from the training 8 views. Then, considering DNGaussian is designed to use depth prior (which is not as accurate as the ground truth mesh) to regularize the optimization process, while vanilla 3DGS has no these regularization, vanilla 3DGS can naturally perform better when the field is initialized very ideally because in this situation DNGaussian may introduce errors from the estimated coarse monocular depth. 
However, I consider it less possible to get such an ideal initialization under sparse-view scenarios. If there really exists such good initialization in your application, I prefer to recommend vanilla 3DGS, or some work else good at maintaining an initialized point cloud structure like Scaffold-GS but not DNGaussian.

**Comment by F011011110100** [comment]
Thank you for your reply! After re-running, I can get a better result of DNGaussian than the source 3DGS. Now the result I get is as follows. 
- paper(1) 21.11(6k steps) 20.46(30k steps)
- paper(2) 22.15(6k steps) 22.05(30k steps)
- paper(3) 22.39(6k steps) 22.15(30k steps)
- 3dgs 21.91(6k steps) 21.54(30k steps)

In addition, the DNGaussian result at 6000 steps is 22.15 or 22.39, which is better than the result of 21.91 of the source 3DGS, but the difference is not large. Is this normal?

**Comment by Fictionarry** [comment]
> Thank you for your reply! After rerun using "random 100,000 pointcloud", I was able to get better DNGaussian results than the source 3DGS. Now I get the following results.
> 
> * paper(1) 21.11(6k steps) 20.46(30k steps)
> * paper(2) 22.15(6k steps) 22.05(30k steps)
> * paper(3) 22.39(6k steps) 22.15(30k steps)
> * 3dgs 21.91(6k steps) 21.54(30k steps)
> 
> In addition, the DNGaussian result at 6000 steps is 22.15 or 22.39, which is better than the result of 21.91 of the source 3DGS, but the difference is not large. Is this normal?

Yes, the difference on Lego is not relatively large. In fact 3DGS can already perform very well on the blender dataset.

**Comment by F011011110100** [comment]
Okay, thank you for your answer.

**Comment by F011011110100** [comment]
Hello, I rechecked and found the problem again. The previous results I ran that were better than 3DGS were not using the "random 100,000 pointcloud", but the default "random 10,000 pointcloud" in DNGaussian. So I re-ran DNGaussian with "random 100,000 pointcloud". But the result is not better than 3DGS. Is there any possible reason for this?
- paper(1) 21.11(6k steps) 20.46(30k steps)(10k)
- paper(2) 22.15(6k steps) 22.05(30k steps)(10k)
- paper(3) 22.39(6k steps) 22.15(30k steps)(10k)
- 3dgs 21.91(6k steps) 21.54(30k steps)(100k)
- paper(1) 20.77(6k steps) 20.70(30k steps)(100k)
- paper(2) 21.51(6k steps) 21.50(30k steps)(100k)
- paper(3) 21.66(6k steps) 21.52(30k steps)(100k)

**Comment by Fictionarry** [comment]
> Hello, I rechecked and found the problem again. The previous results I ran that were better than 3DGS were not using the "random 100,000 pointcloud", but the default "random 10,000 pointcloud" in DNGaussian. So I re-ran DNGaussian with "random 100,000 pointcloud". But the result is not better than 3DGS. Is there any possible reason for this?

Indeed we found a more sparse random point cloud can get better performance in random initialization under sparse-view setting. Our code was developed and adjusted for this situation so you may not see good performance if the point density significantly increases. A similar conclusion can also be found in a later paper https://cvlab-kaist.github.io/RAIN-GS/.  We think it is because a much dense random point cloud would also bring more local minima. It's very possible to see a performance on 3DGS as well if you apply 10,000 points initialization to 3DGS but not use 100,000. Moreover, the performance improvement is not obvious in Lego. 

As the experiment setting you used (test views, training iterations, etc.) is likely not totally the same as ours, we can hardly localize the problem you met. Our Lego tested results following the evaluation setting in the paper are as follows:

DNGaussian:
    {
     "ours_6000": {
      "SSIM": 0.8763810992240906,
      "SSIM_sk": 0.8742926716804504,
      "PSNR": 24.068683624267578,
      "LPIPS": 0.10294689983129501
     }
    }

3DGS original code:
    {
     "ours_30000": {
      "SSIM": 0.8683431148529053,
      "PSNR": 23.719524383544922,
      "LPIPS": 0.110231913626194
     }
    }

**Comment by F011011110100** [comment]
Hi, I saw the experimental setup in your paper, `"We follow the data split used in for the Blender dataset . The 8 input views are selected from the training images,  with IDs 26, 86, 2, 55, 75, 93, 16, 73,  and 8. The 25 test views are sampled evenly from the testing images for evaluation. All images are downsampled 2× to 400 * 400 during the experiment." `My "test" dataset is the full 200 pieces of the original blender dataset lego [test], my "train" dataset is the first 8 pieces of the original lego [train] (IDs 0,1,2,3,4,5,6,7), and I didn't use the [val] folder of the lego dataset.
It could be that my training set and test set caused the PSNR results to be less than ideal.
In addition, the PSNR results of source 3DGS when the initial point cloud number is 10,000 points under my experimental setting are still better than those of DNGaussian when the initial point cloud number is 10,000 points. 
After re-running DNGaussian with an initial number of 100,000 points several times, the result was better than 3DGS only once.
ThereforeSo, I would like to ask if you have tested the results of 3DGS and DNGaussian with a test set consisting of more than 25 images? Also, what was the reason for choosing 25 test sets?
- 3dgs **22.48**(6k steps) **22.21**(30k steps)(10k)
- paper(1) 21.11(6k steps) 20.46(30k steps)(10k)
- paper(2) 22.15(6k steps) 22.05(30k steps)(10k)
- paper(3) 22.39(6k steps) 22.15(30k steps)(10k)
- 3dgs **21.91**(6k steps) **21.54**(30k steps)(100k)
- paper(1) 20.77(6k steps) 20.70(30k steps)(100k)
- paper(2) 21.51(6k steps) 21.50(30k steps)(100k)
- paper(3) 21.66(6k steps) 21.52(30k steps)(100k)
- **paper(4) 21.93(6k steps) 21.84(30k steps)(100k)**

**Comment by F011011110100** [comment]
Okay, thank you very much for your reply!
